### ISSUE TITLE
LAV Splitter crash when opening file with multiple HDMV PGS (SUP) subtitles

### ISSUE THREAD

**Issue by JupiterMu** [Nevcairiel/LAVFilters#41]
I'm currently using Potplayer (64-bit) with LAV filters 0.66.0. However, I experienced constant program crash when I try to open a MKV file with two HDMV/PGS (SUP) tracks. I'm almost certain that the problem is about LAV Splitter and the PGS track, because (1) the problem is solved as soon as I removed the PGS tracks from the video file using MKVMerge 8.4.0 64-bit version, and (2) the problem is also solved without removing the PGS tracks when I switch to another splitter (ffmpeg splitter is the one I tried). I hope this problem can be fixed in future updates because I don't want to switch to another splitter every time I watch this video.

P.S. I just did a further research and removed only one of the two PGS tracks. The crash didn't occur. So it seems that LAV Splitter has trouble handling MULTIPLE PGS tracks.

P.P.S FURTHER research. I examined further by removing the DEFAULT track but leaving the secondary track. Initially the program crashed. Then I examined the flags in MKVToolNix, I found the difference is that the "default track" flag for the previous secondary track (now only and primary track) is marked as "determine automatically". I changed it to "yes" and remux the file, and guess what, it now works. The program doesn't crash now. So further conclude all previous experiments, it seems that LAV Splitter has trouble handling PGS tracks with a "determine automatically" "default track" flag.
![not-working flag](https://cloud.githubusercontent.com/assets/8413180/10995080/5927163e-84b5-11e5-83eb-c29cda3a1210.jpg)
![working flag](https://cloud.githubusercontent.com/assets/8413180/10995081/59d73a28-84b5-11e5-8b0f-f064b3756df7.jpg)

**Comment by Nevcairiel** [comment]
Please provide a sample file that reproduces the crash, its the only way I can fix it.

**Comment by JupiterMu** [comment]
I can't provide one single file, since it comes from a BDRip collection. The link is
https://share.dmhy.org/topics/view/413850_VCB-Studio_Fate_Stay_Night_Unlimited_Blade_Works_10bit_1080p_HEVC_BDRip_Fin.html
The website's main language is in Chinese, but you can download the torrent file by scrolling to the bottom and click the link "[VCB-Studio] Fate/Stay Night Unlimited Blade Works 两季度全 10bit 1080p HEVC BDRip [Fin]". You don't need to download all the files, the file "Fate stay night Unlimited Blade Works [00][Ma10p_1080p][x265_flac].mkv" should be enough. Even then, it's still 1.5GB.
Of course, that is based on the assumption that you can use BT download.

**Comment by Nevcairiel** [comment]
I can't reproduce the problem. The file opens and plays fine here.

**Comment by JupiterMu** [comment]
That was fast.... What player are you using? can you show me a screenshot of the playback info such as splitter, decoders used?

**Comment by Nevcairiel** [comment]
I tried MPC-HC with LAVs default settings. If it crashes, it would produce a crash report, including information in which module it actually crashed. Maybe its a PotPlayer problem.

**Comment by JupiterMu** [comment]
I understand that MPC-HC works, but it uses an internal LAV filters. I'm not sure whether there are any optimizations made by the software itself, but I don't know whether that works for external, original LAV filters?
Besides, the problem is solved when I changed the splitter to ffmpeg, so it doesn't look like the software's problem.
Could you try it on PotPlayer instead? It might take some time to configure, but it uses external filters like LAV installer version instead of internal filter.

**Comment by Nevcairiel** [comment]
I used external LAV Filters of course, not the built-in version. Just because switching to another splitter doesn't show the problem in no way means that its actually LAV that is the problem.

Please provide a crash dump of the crash, then.

**Comment by JupiterMu** [comment]
How do I get that "crash dump"? When I open the video files, the "Potplayer is not working" window pops up and after a few seconds it just says "Potplayer is not working and is closed", something like that. How do I acquire the crash dump?

**Comment by JupiterMu** [comment]
Actually, your last comment did remind me of something. So I completely reset Potplayer's configurations, and try one component a time, from splitter to decoder, to video renderer... Then I found out that it seems my video renderer EVR-CP settings is not compatible with LAV spliter, at least in this case. The "experimental" directx 11.1 mode of EVR seems to be incompatible for this video. I change it back to the original EVR setting and now it works. So it looks like EVR-CP may be incompatible with LAV Splitter occasionally... (Actually this is the first time I've encountered this problem.

Anyway, it appears I've blamed the splitter for everything for the wrong reason. I'm sorry for taking you so much time and trouble.

Just as a side note, may I suggest improving the LAV splitter, particularly with MKV format, further in the future? It's already good right now, but I think it could be better. Haali's splitter does much better in MKV splitting, but it hasn't been updated for a long time and is not compatible with some latest technology. Plus I prefer consistency, from splitter to decoder. So I hope the splitter could be better when dealing with MKV files. Right now there is still some lagging while fast seeking without "jump to keyframe" option.
